tests/components/ActionButton/ToggleActionButton.spec.ts

```diff
@@ -174,8 +185,11 @@ describe('ToggleActionButton', () => {
                     sandbox.reset();
                 });
 
-                it('should not call switchTo when setActivated is called with true', () => {
+                it('should not call switchTo when setActivated is called with true', async () => {
                     testSubject.setActivated(true);
+                    for (let index = 0; index < eventCompletionPromises.length; index++) {
+                        await eventCompletionPromises[index];
+                    }
```

Comment:
Promise.all and Jasmine cannot just cooperate (in fact, Jasmine and promises beyond basics are not working well together at all from what I experienced)